spatial/transform/umeyama.go

```diff
@@ -0,0 +1,146 @@
+// Copyright ©2025 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package transform
+
+import (
+	"errors"
+
+	"gonum.org/v1/gonum/mat"
+	"gonum.org/v1/gonum/stat"
+)
+
+// ErrSVDFailed is returned when a required SVD factorization process fails.
+var ErrSVDFailed = errors.New("transform: SVD factorization failed")
+
+// Umeyama finds the similarity transformation between two sets of points
+// that minimizes the mean squared error between them.
+//
+// The transformation relates two sets of n corresponding points {x_i}
+// and {y_i} as:
+//
+//	y_i ≈ c * R * x_i + t,  i=1,...,n
```

Comment:
Let's use 0-based indexing here so that the notation here matches when we talk about i-th row of X and Y below:
```suggestion
//	y_i ≈ c * R * x_i + t,  i=0,...,n-1
```